File: spring-practice/src/main/java/com/example/springpractice/product/dto/ProductUpdateRequest.java
package com.example.springpractice.product.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.PositiveOrZero;

public record ProductUpdateRequest(
        @NotBlank(message = "상품명은 필수입니다.")
        String name,

        @PositiveOrZero(message = "가격은 0 이상이어야 합니다.")
        int price,

        @PositiveOrZero(message = "재고 수량은 0 이상이어야 합니다.")
        int stockQuantity
) {
}

File: kotlin-practice/src/main/kotlin/com/example/kotlinpractice/product/dto/ProductUpdateRequest.kt
package com.example.kotlinpractice.product.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.PositiveOrZero

data class ProductUpdateRequest(
    @field:NotBlank(message = "상품명은 필수입니다.")
    val name: String,

    @field:PositiveOrZero(message = "가격은 0 이상이어야 합니다.")
    val price: Int,

    @field:PositiveOrZero(message = "재고 수량은 0 이상이어야 합니다.")
    val stockQuantity: Int
)
